tinygrad/engine/schedule.py

```diff
@@ -111,7 +109,7 @@ def _recurse_lb(buf:LazyBuffer, realizes:Dict[LazyBuffer, None], allbufs:Dict[La
       if len(buf.st.views) == 1 and buf.st.views[-1].mask and all_int(buf.base.st.shape) and \
           prod(buf.base.st.shape) >= prod([y-x for x,y in buf.st.views[-1].mask]):
         simple_pads.add(buf.base)
-      else:
+      elif all([x.op not in ReduceOps for x in buf.base.srcs if hasattr(x, "op")]):
```

Comment:
what if buf isn't the only child of x?
```py
x = Tensor.empty(32)
r0 = x.mean(axis=0, keepdim=True)
r1 = (x - r0).sum(axis=0).div(2)
e = r0 + r1
e.realize()
```

https://tiny-tools-client.vercel.app/?id=7b4dc4f7c0c34dcb94e1d82639cc3180

<img width="220" alt="graph" src="https://github.com/tinygrad/tinygrad/assets/77887910/ac435f74-cd80-41ad-958b-d5cb905734c3">